### ISSUE TITLE
 Received the following error upon using dist_train with 8 GPUs._pickle.UnpicklingError: pickle data was truncated.

### ISSUE THREAD

**Issue by khurramHashmi** [open-mmlab/mmtracking#350]
Thanks for your error report and we appreciate it a lot.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version.

**Describe the bug**

**Reproduction**

1. What command or script did you run?

```none
A placeholder for the command.
```

2. Did you make any modifications on the code or config? Did you understand what you have modified?
3. What dataset did you use and what task did you run?

**Environment**

1. Please run `python mmtrack/utils/collect_env.py` to collect necessary environment information and paste it here.
2. You may add addition that may be helpful for locating the problem, such as
    - How you installed PyTorch [e.g., pip, conda, source]
    - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

**Error traceback**
If applicable, paste the error trackback here.

```none
A placeholder for trackback.
```

**Bug fix**
If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by GT9505** [comment]
Please use [error_report.md](https://github.com/open-mmlab/mmtracking/blob/master/.github/ISSUE_TEMPLATE/error-report.md) to report bug.

**Comment by khurramHashmi** [comment]
---
name: Error report
about: Create a report to help us improve
title: 'UnpicklingError: pickle data was truncated.'
labels: ''
assignees: ''

---


**Checklist**
I have searched related issues but cannot get the expected help.
The bug has not been fixed in the latest version.

By using any configuration with dist_train on 8 GPUs will give the following error:
UnpicklingError: pickle data was truncated

**Reproduction**

1. Command : CUDA_VISIBLE_DEVICES=0,1,2,3 PORT=29502 tools/dist_test.sh configs/vid/temporal_roi_align/selsa_troialign_faster_rcnn_r50_dc5_7e_imagenetvid.py 4 --checkpoint selsa_troialign_faster_rcnn_r50_dc5_7e_imagenetvid_20210820_162714-939fd657.pth --out results_r50.pkl --eval bbox



2. Did you make any modifications on the code or config? Did you understand what you have modified?
No modifications
3. What dataset did you use and what task did you run?
ImageNet Vid val set.

**Environment**

sys.platform: linux
Python: 3.7.9 (default, Aug 31 2020, 12:42:55) [GCC 7.3.0]
CUDA available: True
GPU 0: GeForce GTX 1080 Ti
GPU 1: GeForce GT 1030
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.2, V10.2.89
GCC: gcc (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0
PyTorch: 1.7.0
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.2 Product Build 20200624 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v1.6.0 (Git Hash 5ef631a030a6f73131c77892041042805a06064f)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS=MKL, BUILD_TYPE=Release, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DUSE_VULKAN_WRAPPER -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, USE_CUDA=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.8.0
OpenCV: 4.5.1
MMCV: 1.3.8
MMCV Compiler: GCC 7.3
MMCV CUDA Compiler: 10.2
MMTracking: 0.8.0+4d68645

**Error traceback**
none

**Comment by khurramHashmi** [comment]
@GT9505 Could you please respond?

**Comment by GT9505** [comment]
Please fill the `Error traceback`.
Besides, what command did you run? In the `Reproduction`, you use `tools/dist_test.sh` with 4 gpus, while in the `Checklist`, you use `dist_train`  on 8 GPUs to get the error.
Please make the error information more clearly.

Generally, `UnpicklingError: pickle data was truncated` means that the file is damaged when you use `pickle.load` to load the file.

**Comment by khurramHashmi** [comment]
I have corrected the command.
The pickle file seems to be fine since it totally works with 4 GPUs. The problem initiates on distributed training with 8 GPUs.

**Comment by GT9505** [comment]
This is very weird.
As mentioned in [here](https://github.com/open-mmlab/mmtracking/issues/333#issuecomment-983923458), when setting `num_worker_per_gpu=1`, you can run distributed training with 8 GPUs. While setting `num_worker_per_gpu=2`, distributed training on 8 gpus will raise the error.

**Comment by khurramHashmi** [comment]
Indeed, it is weird. Even the weirder part is that it used to work before with GPUs. 
Also, I have tried with `num_worker_per_gpu=1`. It did not work.
